control/HTTP.php

```diff
@@ -318,139 +319,117 @@ public static function register_etag($etag) {
-		// Development sites have frequently changing templates; this can get stuffed up by the code
-		// below.
-		if(Director::isDev()) $cacheAge = 0;
-
 		// The headers have been sent and we don't have an SS_HTTPResponse object to attach things to; no point in
 		// us trying.
-		if(headers_sent() && !$body) return;
-
-		// Populate $responseHeaders with all the headers that we want to build
-		$responseHeaders = array();
+		if(headers_sent() && !$body) {
+			return;
+		}
 
 		$config = Config::inst();
-		$cacheControlHeaders = Config::inst()->get('HTTP', 'cache_control');
 
+		// Development sites have frequently changing templates; this can get stuffed up by the code
+		// below.
+		if(Director::isDev()) {
+			HTTPCacheControl::inst()->disableCaching();
+		}
 
-		// currently using a config setting to cancel this, seems to be so that the CMS caches ajax requests
-		if(function_exists('apache_request_headers') && $config->get(get_called_class(), 'cache_ajax_requests')) {
+		// Populate $responseHeaders with all the headers that we want to build
+		$responseHeaders = array();
+		$cacheControlHeaders = Config::inst()->get(__CLASS__, 'cache_control');
+
+		// if no caching ajax requests, disable ajax if is ajax request
+		// why are we using apache_request_headers here when we use `$_SERVER` later to inspect request headers?
+		if (!$config->get(__CLASS__, 'cache_ajax_requests') && function_exists('apache_request_headers')) {
 			$requestHeaders = array_change_key_case(apache_request_headers(), CASE_LOWER);
 
-			if(isset($requestHeaders['x-requested-with']) && $requestHeaders['x-requested-with']=='XMLHttpRequest') {
-				$cacheAge = 0;
+			if (array_key_exists('x-requested-with', $requestHeaders) && strtolower($requestHeaders['x-requested-with']) == 'xmlhttprequest') {
+				HTTPCacheControl::inst()->disableCaching();
 			}
 		}
 
-		if($cacheAge > 0) {
-			$cacheControlHeaders['max-age'] = self::$cache_age;
-
-			// Set empty pragma to avoid PHP's session_cache_limiter adding conflicting caching information,
-			// defaulting to "nocache" on most PHP configurations (see http://php.net/session_cache_limiter).
-			// Since it's a deprecated HTTP 1.0 option, all modern HTTP clients and proxies should
-			// prefer the caching information indicated through the "Cache-Control" header.
-			$responseHeaders["Pragma"] = "";
-
-			// To do: User-Agent should only be added in situations where you *are* actually
-			// varying according to user-agent.
-			$vary = $config->get('HTTP', 'vary');
-			if ($vary && strlen($vary)) {
-				$responseHeaders['Vary'] = $vary;
+		$vary = $config->get(__CLASS__, 'vary');
+		if ($vary && strlen($vary)) {
+			if ($body) {
+				$currentVary = $body->getHeader('Vary');
 			}
-		}
-		else {
-			if($body) {
-				// Grab header for checking. Unfortunately HTTPRequest uses a mistyped variant.
-				$contentDisposition = $body->getHeader('Content-disposition', true);
+			if ($currentVary) {
+				$vary = $currentVary . ', ' . $vary;
```

Comment:
This could lead to double ups (from the specific response and the generic config), but that's acceptable - I assume it's unlikely to throw off HTTP caches. So before generic config would override specific `Vary` headers on the response, which makes that feature useless. Now it appends both. Ideally you could opt out of certain `Vary` headers set in config (e.g. instruct to vary on cookie, *apart* from specific responses). But that's quite specialised, can't think of a great use case.

Reply:
Making this unique shouldn't be too tough; I'll look at adding that in

Reply:
Made this unique for now given the current approach, but this may change with some refactors